### PR TITLE
Dynamic offsets

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: libnopegl/src/pipeline_compat.h [hunk 1]
```diff
@@ -41,6 +41,7 @@ int ngli_pipeline_compat_update_uniform(struct pipeline_compat *s, int32_t index
 int ngli_pipeline_compat_update_texture(struct pipeline_compat *s, int32_t index, const struct texture *texture);
 void ngli_pipeline_compat_update_texture_info(struct pipeline_compat *s, const struct pgcraft_texture_info *info);
 int ngli_pipeline_compat_update_buffer(struct pipeline_compat *s, int32_t index, const struct buffer *buffer, int offset, int size);
+int ngli_pipeline_compat_update_dynamic_offsets(struct pipeline_compat *s, uint32_t *offsets, size_t nb_offsets);
```
Comment (ubitux) [inline libnopegl/src/pipeline_compat.h:44]: can we have `const uint32_t *offsets`?

**Review by ubitux — APPROVED** [review]
Thanks, should be fine apart from my comment

File: libnopegl/src/pipeline_compat.h (hunk 1 above)
Comment (mbouron) [reply libnopegl/src/pipeline_compat.h:44]: Done.

Comment (ubitux) [reply libnopegl/src/pipeline_compat.h:44]: You did it for the pipeline but not the compat layer

File: libnopegl/src/pipeline.c [hunk 2]
```diff
@@ -95,6 +95,19 @@ int ngli_pipeline_init(struct pipeline *s, const struct pipeline_params *params)
     if (ret < 0)
         return ret;
 
+    size_t nb_uniform_buffers_dynamic = 0;
+    size_t nb_storage_buffers_dynamic = 0;
+    for (size_t i = 0; i < s->layout.nb_buffer_descs; i++) {
+        const struct pipeline_resource_desc *desc = &s->layout.buffer_descs[i];
+        if (desc->type == NGLI_TYPE_UNIFORM_BUFFER_DYNAMIC)
+            nb_uniform_buffers_dynamic++;
+        if (desc->type == NGLI_TYPE_STORAGE_BUFFER_DYNAMIC)
```
Comment (ubitux) [inline libnopegl/src/pipeline.c:104]: could be an `else if` btw

Comment (mbouron) [reply libnopegl/src/pipeline.c]: Fixed. Thanks.

File: libnopegl/src/pipeline_compat.h (hunk 1 above)
Comment (mbouron) [reply libnopegl/src/pipeline_compat.h]: Done.

**Review by ubitux — APPROVED** [review]
LGTM (you might want to add some assert wrt limits vs `NGLI_MAX_*_BUFFERS_DYNAMIC`)

**Pull request merged**